### DISCUSSION TITLE
Curious about sync storage

### DISCUSSION THREAD

**Discussion by marcustyphoon** [AprilSylph/XKit-Rewritten#188] [Lounge]
Hallo! 👋

Not important, but I was curious about the motivations behind removing cloud sync besides the storage key length and quantity limitations.  https://github.com/AprilSylph/XKit-Rewritten/commit/04982538832e202c221b52c037f36f3e93114c62 mentions browser variance and I thought I vaguely remember you discussing it somewhere, but I can't remember where :D

(by the way, are you encouraging of/indifferent to/disapproving of/etc using the repo's discussions tab for... well, kind of like a discord channel's general chat?)

**Comment by AprilSylph** [comment]
> (by the way, are you encouraging of/indifferent to/disapproving of/etc using the repo's discussions tab for... well, kind of like a discord channel's general chat?)

i encourage it!

---

Instead of giving a long, complete, and complicated answer as I just was attempting to... it boils down to not every browser supporting it (Opera, Firefox for Android) or supporting it properly (Safari), the complexity of ensuring the API is used within its limits, and the paradox of designing a user-friendly interface for something that outright rejects the needs of a notable subset of users (one which I belong to). I don't want to rule using `storage.sync` out forever, but it comes with some high hurdles which I don't feel are worth jumping yet, if ever.
